lib/net/networking_engine.js

```diff
@@ -241,14 +241,13 @@ shaka.net.NetworkingEngine = class extends shaka.util.FakeEventTarget {
     // undefined.  To make it easier for application developers, we will fill
     // them in with defaults if necessary.
     //
-    // We clone retryParameters and uris so that if a filter modifies the
+    // We clone retryParameters so that if a filter modifies the
     // request, it doesn't contaminate future requests.
     request.method = request.method || 'GET';
     request.headers = request.headers || {};
     request.retryParameters = request.retryParameters ?
         ObjectUtils.cloneObject(request.retryParameters) :
         shaka.net.NetworkingEngine.defaultRetryParameters();
-    request.uris = ObjectUtils.cloneObject(request.uris);
```

Comment:
We need the final original URI of the request in the maifestUris_ to replace it with the redirected one. I think these changes are safe as with the current approach, every time the manifest URI changes (for example in request filters) it will be added to the manifestUris_ array.

Reply:
I don't think app-level filters should be able to modify a private array inside the library.  That could break all kinds of expectations.  And it looks like you don't even need this, since you have the dash parser updating the array when it gets a response.

Reply:
The problem is that we need to have the exact request URL in the manifest URI array in order to be able to find the index of it and replace this URI with the new one.

In case when the request URI was changed in request filters, the manifestUris_ array will not include the exact uri of the request,  and _this.manifestUris.indexOf(response.originalUri)_ will be -1 and we will not be able to understand what url needs to be replaced. 

For example scenario:
1.this.manifestUris_ = ['url']
2. in request filter we change the uris[0] it to  'url-after-filter'
3. this url redirected with 302 code to 'url-after-redirect'
4. so without array modification, we will receive the next after awaiting request 
```
this.manifestUris_ = ['url']
response.uri = 'url-after-redirect'
response.originalUri = 'url-after-filter'
```